server/middlewares/index.js

```diff
@@ -18,16 +18,16 @@ exports.isAuth = async (req, res, next) => {
 exports.userById = (req, res, next, id) => {
```

Comment:
Just a heads up: if there are other cases in which someone would use this userById without needing the projects and investments and the associated projects of the investments, then this will be an unnecessarily costly operation. In the case that this will always only be used to get all of this data, I think it might best be renamed so that it's purpose is clear.

Reply:
Yeah, I may either rename it or remove it. I'll have to think about it more and then see how it is when I resubmit it for review later today.

Reply:
The thing is that it's a cool middleware. I wonder if it can be made flexible -- responding to additional params.

Reply:
That's a great idea for the params. I wonder if that could help change the calls. I also know that getUser method is different from this userById middleware. Because for getUser, when we get the user info we will want their projects and investments. userById could probably just be used to see if the user exists and their basic info

Reply:
Oooh. I like that. Could really simplify things.